### PR TITLE
docs(aio): fix lots of broken links

### LINKED ISSUE #16317: Broken link on the quick start page
screen shot 2017-04-25 at 4 51 10 pm

I don't know what this link should point at, I no longer see the learning-angular.md file anywhere.

### PR DESCRIPTION
**Pull request by Foxandxss**
I haven't touch neither the `index.md` (it needs rewrite), neither the `style-guide` because it is heavily broken and the toh-pt6 has a link to next step which is the "learning path" that we deleted so we need to decide where should it point.

Closes #16317

### CODE REVIEW

File: aio/content/guide/dependency-injection.md [hunk 1]
```diff
@@ -611,7 +611,7 @@ You call that property within the `getHeroes()` method when anyone asks for hero
 
 ### Why _@Injectable()_?
 
-**<a href="../api/core/index/Injectable-decorator.html">@Injectable()</a>** marks a class as available to an
+**[@Injectable](api/core/Injectable)** marks a class as available to an
```
Comment (petebacondarwin) [inline aio/content/guide/dependency-injection.md]: I think this should be:

```
**[`@Injectable()`](api/core/Injectable)** marks a class as available to an
```

i.e. it should have the parenthesis to indicate the usage is as a call, and also wrapped in single backticks to make it look like code.

File: aio/content/guide/dependency-injection.md [hunk 2]
```diff
@@ -671,7 +671,7 @@ like `HeroesComponent`. So why doesn't `HeroesComponent` have
 You *can* add it if you really want to. It isn't necessary because the
 `HeroesComponent` is already marked with `@Component`, and this
 decorator class (like `@Directive` and `@Pipe`, which you learn about later)
-is a subtype of <a href="../api/core/index/Injectable-decorator.html">@Injectable()</a>.  It is in
+is a subtype of [@Injectable](api/core/Injectable).  It is in
```
Comment (petebacondarwin) [inline aio/content/guide/dependency-injection.md]: backticks around this one too

File: aio/content/guide/dependency-injection.md [hunk 3]
```diff
@@ -692,7 +692,7 @@ the compiler adds the metadata to the generated JavaScript
 for _every class with at least one decorator_.
 
 While any decorator will trigger this effect, mark the service class with the
-<a href="../api/core/index/Injectable-decorator.html">@Injectable()</a> decorator
+[@Injectable](api/core/Injectable) decorator
```
Comment (petebacondarwin) [inline aio/content/guide/dependency-injection.md]: backticks and parentheses here too

File: aio/content/guide/dependency-injection.md [hunk 4]
```diff
@@ -1147,7 +1147,7 @@ There is no interface type information left for Angular to find at runtime.
 ### _InjectionToken_
 
 One solution to choosing a provider token for non-class dependencies is
-to define and use an <a href="../api/core/index/InjectionToken-class.html"><b>InjectionToken</b></a>.
+to define and use an [*InjectionToken*](api/core/InjectionToken).
```
Comment (petebacondarwin) [inline aio/content/guide/dependency-injection.md]: backticks rather than bold?

File: aio/content/guide/pipes.md [hunk 5]
```diff
@@ -85,7 +85,7 @@ They are all available for use in any template.
 
 
 
-Read more about these and many other built-in pipes in the [pipes topics](api/#!?query=pipe) of the
+Read more about these and many other built-in pipes in the [pipes topics](api?query=pipe) of the
```
Comment (petebacondarwin) [inline aio/content/guide/pipes.md]: should this link use the `type=pipe` rather than `query=pipe` param?

Comment (petebacondarwin) [inline aio/content/guide/pipes.md]: or is the idea that the query also picks up the `Pipe` class itself?

File: aio/content/guide/pipes.md [hunk 6]
```diff
@@ -621,7 +621,7 @@ transformations. Use them like styles, dropping them
 into your template's expressions to enrich the appeal and usability
 of your views.
 
-Explore Angular's inventory of built-in pipes in the [API Reference](api/#!?query=pipe).
+Explore Angular's inventory of built-in pipes in the [API Reference](api?query=pipe).
```
Comment (petebacondarwin) [inline aio/content/guide/pipes.md]: Pretty sure this should be `type=pipe`

File: aio/content/guide/quickstart.md [hunk 7]
```diff
@@ -73,7 +73,7 @@ In the example, change the component class's `name` property from `'Angular'` to
 
 ### Next step
 
-Start [**learning Angular**](guide/guide/learning-angular "Learning Angular").
+Start the [**tutorial**](tutorial "Tour of Heroes tutorial").
```
Comment (petebacondarwin) [inline aio/content/guide/quickstart.md]: :heart:

**Review by petebacondarwin — COMMENTED** [review]
A few comments. I appreciate that you are keeping the local style the same but at some point we need to do a flush through and ensure that all API links are more consistent - currently they use a mixture of backticks, bold and emphasis.

File: aio/content/guide/dependency-injection.md (hunk 1 above)
Comment (Foxandxss) [inline aio/content/guide/dependency-injection.md]: Why do you think so? I don't see them in the original.

File: aio/content/guide/quickstart.md (hunk 7 above)
Comment (Foxandxss) [inline aio/content/guide/quickstart.md]: Any problem here?

File: aio/content/guide/dependency-injection.md (hunk 1 above)
Comment (petebacondarwin) [inline aio/content/guide/dependency-injection.md]: But still `@Injectable()` make it callable.

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**